lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -324,13 +399,15 @@ def message_from(succeeded, parsed_response)
         parsed_response['description']
       end
 
-      def authorization_from(action, parsed_response)
-        action == :store ? vaulted_shopper_id(parsed_response) : parsed_response['transaction-id']
+      def authorization_from(action, parsed_response, payment_method_details)
+        action == :store ? vaulted_shopper_id(parsed_response, payment_method_details) : parsed_response['transaction-id']
       end
 
-      def vaulted_shopper_id(parsed_response)
+      def vaulted_shopper_id(parsed_response, payment_method_details)
         return nil unless parsed_response['content-location-header']
-        parsed_response['content-location-header'].split('/').last
+        vaulted_shopper_id = parsed_response['content-location-header'].split('/').last
+        vaulted_shopper_id += "|#{payment_method_details.payment_method_type}" if payment_method_details.alt_transaction?
+        vaulted_shopper_id
```

Comment:
I might just be confused, but shouldn't this mean that for a purchase with a stored card, the token is added to the request as `[id]|card`? I'd think it would need to get split out, but apparently the `test_successful_purchase_using_stored_card` remote test passes?

Reply:
Oh, no, because it's not an `alt_transaction?` on store. NM!